### DISCUSSION TITLE
Actions output

### DISCUSSION THREAD

**Discussion by cstriuli** [filamentphp/filament#11465] [Help]
### Package

Actions

### Package Version

v.3.0.0

### How can we help you?

Is it possible for Actions to output the contents of `Artisan::output`, similar to what you'd observe on Forge or other Laravel tools? Essentially, this would display the terminal output.

Why? Because I'm running an action that invokes `Artisan` command, it could take some time (around 1-3 minutes). Therefore, I was considering displaying the output directly in the same modal, but I couldn't find any resources on this.

```php
Tables\Actions\Action::make('Migrate data from API')
    ->form([...])
    ->action(function (array $data) {
        Artisan::call('migrate:data', $data);
       
       
       // :(
        Notification::make()
            ->body(Artisan::output())
            ->duration(100000)
            ->info()
            ->send();
    })
```

Example of output:
![image](https://github.com/filamentphp/filament/assets/3141550/9056d6a6-1598-4d25-8bd7-02fd84215216)

**Comment by ArtMin96** [comment]
Something like this. you can test this and implement your requirements

```php
use Exception;
use Illuminate\Support\Facades\Artisan;
use Symfony\Component\Console\Output\BufferedOutput;

public function method()
{
    $outputLog = new BufferedOutput;
    return $this->runCommand($outputLog, $options)
}

protected function runCommand(BufferedOutput $outputLog, array  $options)
{
    try {
        Artisan::call('my-artisan:command', $options, $outputLog);
    } catch (Exception $e) {
        return $this->response($e->getMessage(), $outputLog);
    }

    return $outputLog;
}

protected function response(string $message, BufferedOutput $outputLog): array
{
    return [
        'status' => 'error',
        'message' => $message,
        'dbOutputLog' => $outputLog->fetch(),
    ];
}
```

And somewhere
```php
->method()
```

And you need response streaming
[Streaming responses in Livewire](https://livewire.laravel.com/docs/wire-stream#streaming-chat-bot-responses)
